File: java/src/main/java/LdProofs.java
import com.google.protobuf.InvalidProtocolBufferException;
import trinsic.okapi.Proofs;

public class LdProofs extends OkapiNative {
    public static Proofs.CreateProofResponse createProof(Proofs.CreateProofRequest request) throws DidException, InvalidProtocolBufferException {
        OkapiByteBuffer.ByValue requestBuffer = messageToBuffer(request);
        OkapiByteBuffer responseBuffer = new OkapiByteBuffer();
        ExternError errBuffer = new ExternError();
        getNativeLibrary().ldproofs_create_proof(requestBuffer, responseBuffer, errBuffer);
        errBuffer.RaiseError();
        return Proofs.CreateProofResponse.parseFrom(bufferToByteArray(responseBuffer));
    }

    public static Proofs.VerifyProofResponse verifyProof(Proofs.VerifyProofRequest request) throws DidException, InvalidProtocolBufferException {
        OkapiByteBuffer.ByValue requestBuffer = messageToBuffer(request);
        OkapiByteBuffer responseBuffer = new OkapiByteBuffer();
        ExternError errBuffer = new ExternError();
        getNativeLibrary().ldproofs_verify_proof(requestBuffer, responseBuffer, errBuffer);
        errBuffer.RaiseError();
        return Proofs.VerifyProofResponse.parseFrom(bufferToByteArray(responseBuffer));
    }
}

File: ruby/lib/ld_proofs.rb
# frozen_string_literal: true

# LdProofs module
module LdProofs
  def self.create(request)
    Okapi.verify_type(request, Proofs_V1::CreateProofRequest)
    Okapi.ffi_call('ldproofs_create_proof', request, Proofs_V1::CreateProofResponse)
  end

  def self.verify(request)
    Okapi.verify_type(request, Proofs_V1::VerifyProofRequest)
    Okapi.ffi_call('ldproofs_verify_proof', request, Proofs_V1::VerifyProofResponse)
  end
end
